packages/block-editor/src/components/contrast-checker/index.js

```diff
@@ -1,125 +1,139 @@
+import { speak } from '@wordpress/a11y';
+import { Notice } from '@wordpress/components';
+
 /**
  * External dependencies
  */
 import { colord, extend } from 'colord';
 import namesPlugin from 'colord/plugins/names';
 import a11yPlugin from 'colord/plugins/a11y';
 
-/**
- * WordPress dependencies
- */
-import { speak } from '@wordpress/a11y';
-import { __ } from '@wordpress/i18n';
-import { Notice } from '@wordpress/components';
-import { useEffect } from '@wordpress/element';
-
 extend( [ namesPlugin, a11yPlugin ] );
 
-function ContrastCheckerMessage( {
-	colordBackgroundColor,
-	colordTextColor,
-	backgroundColor,
-	textColor,
-	shouldShowTransparencyWarning,
-} ) {
-	let msg = '';
-	if ( shouldShowTransparencyWarning ) {
-		msg = __( 'Transparent text may be hard for people to read.' );
-	} else {
-		msg =
-			colordBackgroundColor.brightness() < colordTextColor.brightness()
-				? __(
-						'This color combination may be hard for people to read. Try using a darker background color and/or a brighter text color.'
-				  )
-				: __(
-						'This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.'
-				  );
-	}
-
-	// Note: The `Notice` component can speak messages via its `spokenMessage`
-	// prop, but the contrast checker requires granular control over when the
-	// announcements are made. Notably, the message will be re-announced if a
-	// new color combination is selected and the contrast is still insufficient.
-	useEffect( () => {
-		const speakMsg = shouldShowTransparencyWarning
-			? __( 'Transparent text may be hard for people to read.' )
-			: __( 'This color combination may be hard for people to read.' );
-		speak( speakMsg );
-	}, [ backgroundColor, textColor ] );
-
-	return (
-		<div className="block-editor-contrast-checker">
-			<Notice
-				spokenMessage={ null }
-				status="warning"
-				isDismissible={ false }
-			>
-				{ msg }
-			</Notice>
-		</div>
-	);
-}
-
 function ContrastChecker( {
 	backgroundColor,
 	fallbackBackgroundColor,
 	fallbackTextColor,
+	fallbackLinkColor,
 	fontSize, // font size value in pixels
 	isLargeText,
 	textColor,
+	linkColor,
 	enableAlphaChecker = false,
 } ) {
-	if (
-		! ( backgroundColor || fallbackBackgroundColor ) ||
-		! ( textColor || fallbackTextColor )
-	) {
+	const currentBackgroundColor = backgroundColor || fallbackBackgroundColor;
+
+	// Must have a background color.
+	if ( ! currentBackgroundColor ) {
 		return null;
 	}
-	const colordBackgroundColor = colord(
-		backgroundColor || fallbackBackgroundColor
-	);
-	const colordTextColor = colord( textColor || fallbackTextColor );
-	const textColorHasTransparency = colordTextColor.alpha() < 1;
+
+	const currentTextColor = textColor || fallbackTextColor;
+	const currentLinkColor = linkColor || fallbackLinkColor;
+
+	// Must have at least one text color.
+	if ( ! currentTextColor && ! currentLinkColor ) {
+		return null;
+	}
+
+	const textColors = [
```

Comment:
This could, in future, be a model for this component's props as illustrated in 00e5e31f402c7231af995afa25a38c42a61d40a3

Then we'd be able to test `n` number of foreground colors against a single background color.

Reply:
Agreed, this could form a basis for a future approach. There could be some edge cases though when it comes to hover or focus states etc.